sklearn/preprocessing/tests/test_data.py

```diff
@@ -918,172 +894,126 @@ def _check_identity_scalers_attributes(scaler_1, scaler_2):
     assert scaler_1.n_samples_seen_ == scaler_2.n_samples_seen_
 
 
-def test_scaler_return_identity():
+@pytest.mark.parametrize("sparse_container", CSC_CONTAINERS + CSR_CONTAINERS)
+def test_scaler_return_identity(sparse_container):
     # test that the scaler return identity when with_mean and with_std are
     # False
     X_dense = np.array([[0, 1, 3], [5, 6, 0], [8, 0, 10]], dtype=np.float64)
-    X_csr = sparse.csr_matrix(X_dense)
-    X_csc = X_csr.tocsc()
+    X_sparse = sparse_container(X_dense)
 
     transformer_dense = StandardScaler(with_mean=False, with_std=False)
     X_trans_dense = transformer_dense.fit_transform(X_dense)
-
-    transformer_csr = clone(transformer_dense)
-    X_trans_csr = transformer_csr.fit_transform(X_csr)
-
-    transformer_csc = clone(transformer_dense)
-    X_trans_csc = transformer_csc.fit_transform(X_csc)
-
-    assert_allclose_dense_sparse(X_trans_csr, X_csr)
-    assert_allclose_dense_sparse(X_trans_csc, X_csc)
     assert_allclose(X_trans_dense, X_dense)
 
-    for trans_1, trans_2 in itertools.combinations(
-        [transformer_dense, transformer_csr, transformer_csc], 2
```

Comment:
I think this test was previously also testing the csr vs csc combination. Are we still doing that?

Reply:
I think this is now testing csr vs dense and csc vs dense. If they both pass, should it imply that csr vs csc will also pass or no? Maybe the tolerance of `assert_allclose` and `assert_allclose_dense_sparse` will be loosen a bit, not sure if that matters.

Reply:
I think it makes sense that both are being compared with dense.